docker/swarmkit
# Add Dockerfile and containerized build

**- What I did**
Adds a Dockerfile for the swarmkit project, to easily get off the ground. Modifies the Makefile to make intelligent use of Docker.

**- How I did it**
Modifies the Makefile to have two paths: containerized.mk, which builds the docker image and forwards any make targets to a container, and direct.mk, which encompasses the old Makefile's workflow.

By default, nothing will run inside a container. Set the environment variable `SWARMKIT_USE_CONTAINER` to use dockerized making.

**- How to test it**
Set `SWARMKIT_USE_CONTAINER` and verify that your favorite make targets all work
